lib/ansible/modules/source_control/git.py

```diff
@@ -371,9 +371,10 @@ def write_ssh_wrapper():
 
 def set_git_ssh(ssh_wrapper, key_file, ssh_opts):
 
-    if os.environ.get("GIT_SSH"):
-        del os.environ["GIT_SSH"]
+    # git_ssh_command will override git_ssh, so only older git needs it
     os.environ["GIT_SSH"] = ssh_wrapper
+    # using shell to avoid 'noexec' issues if module temp dir is located in such a mount
+    os.environ["GIT_SSH_COMMAND"] = ' '.join([os.environ.get('SHELL', '/bin/sh'), ssh_wrapper])
```

Comment:
The `join` here seems a little unnecessary, and the performance is worse than a printf like string format or using `.format`

Although it is just 1 line, and not repeated, so that's just a little bit of a nitpick.

Reply:
Updated to use percent formatting